sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -228,6 +229,101 @@ abstract class QueryTest extends PlanTest {
     assert(query.queryExecution.executedPlan.missingInput.isEmpty,
       s"The physical plan has missing inputs:\n${query.queryExecution.executedPlan}")
   }
+
+  /** A single SQL query's output. */
+  protected trait QueryTestOutput {
```

Comment:
Now all the changes in `QueryTest` is only used in `SQLQueryTestSuite`. Shall we move these changes to `SQLQueryTestSuite`?